src/Symfony/Component/VarDumper/Caster/DateCaster.php

```diff
@@ -79,4 +79,31 @@ public static function castTimeZone(\DateTimeZone $timeZone, array $a, Stub $stu
 
         return $filter & Caster::EXCLUDE_VERBOSE ? $z : $z + $a;
     }
+
+    public static function castPeriod(\DatePeriod $p, array $a, Stub $stub, $isNested, $filter)
+    {
+        if (defined('HHVM_VERSION_ID') || \PHP_VERSION_ID < 50605) {
+            return $a;
+        }
+
+        $dates = array();
+        if (\PHP_VERSION_ID >= 70107) { // see https://bugs.php.net/bug.php?id=74639
+            $format = 'Y-m-d'.('000000' === $p->getStartDate()->format('His') && '000' === $p->getDateInterval()->format('%h%i%s') ? '' : ' H:i:s');
+            foreach (clone $p as $i => $d) {
+                $dates[] = sprintf('%s) %s', $i + 1, $d->format($format));
```

Comment:
Mmh, an overview of dates period can be very useful I think... Let's wait for other opinions ;)